fabric-ingredient-api-v1/src/main/java/net/fabricmc/fabric/api/ingredient/v1/DefaultCustomIngredients.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.ingredient.v1;
+
+import java.util.Objects;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.nbt.NbtHelper;
+import net.minecraft.recipe.Ingredient;
+
+import net.fabricmc.fabric.impl.ingredient.builtin.AndIngredient;
+import net.fabricmc.fabric.impl.ingredient.builtin.DifferenceIngredient;
+import net.fabricmc.fabric.impl.ingredient.builtin.NbtIngredient;
+import net.fabricmc.fabric.impl.ingredient.builtin.OrIngredient;
+
+/**
+ * Factory methods for the custom ingredients directly provided by fabric.
+ */
+public final class DefaultCustomIngredients {
+	/**
+	 * Create an ingredient that matches what its sub-ingredients all match.
+	 *
+	 * <p>The JSON format is as follows:
+	 * <pre>{@code
+	 * {
+	 *     "fabric:type": "fabric:and",
+	 *     "ingredients": [
+	 *         // sub-ingredient 1,
+	 *         // sub-ingredient 2,
+	 *         // etc...
+	 *     ]
+	 * }
+	 * }</pre>
+	 */
+	public static Ingredient and(Ingredient... ingredients) {
+		for (Ingredient ing : ingredients) Objects.requireNonNull(ing);
+
+		return new AndIngredient(ingredients).toVanilla();
+	}
+
+	/**
+	 * Create an ingredient that matches what any of its sub-ingredients matches.
+	 *
+	 * <p>The JSON format is as follows:
+	 * <pre>{@code
+	 * {
+	 *     "fabric:type": "fabric:or",
+	 *     "ingredients": [
+	 *         // sub-ingredient 1,
+	 *         // sub-ingredient 2,
+	 *         // etc...
+	 *     ]
+	 * }
+	 * }</pre>
+	 */
+	public static Ingredient or(Ingredient... ingredients) {
+		for (Ingredient ing : ingredients) Objects.requireNonNull(ing);
+
+		return new OrIngredient(ingredients).toVanilla();
+	}
+
+	/**
+	 * Create an ingredient that matches if its base ingredient matches, and its subtracted ingredient does NOT match.
```

Comment:
```suggestion
	 * Creates an ingredient that matches if its base ingredient matches, and its subtracted ingredient <strong>does not</strong> match.
```

Why CAPS when we can **strong**?